src/condor_schedd.V6/qmgmt.cpp

```
@@ -4719,9 +4719,18 @@ SetAttribute(int cluster_id, int proc_id, const char *attr_name,
 	if (attr_id == idATTR_OWNER)
 	{
 		// User and Owner can't be set in ordinary job ads. Force those into
-		// the cluster ad.
-		if (cluster_id > 0 && proc_id >= 0) {
-			return SetAttribute(cluster_id, -1, attr_name, attr_value, flags, err);
+		// the cluster ad if proc_id==0 (handle old c-gahp and job router)
+		// and reject otherwise.
+		if (cluster_id > 0) {
+			if (proc_id == 0) {
+				return SetAttribute(cluster_id, -1, attr_name, attr_value, flags, err);
+			} else {
+				dprintf(D_ALWAYS, "ERROR SetAttribute violation: "
+					"Owner is being set in a proc ad\n");
+				if (err) err->pushf("QMGMT", EACCES, "Owner set in proc ad");
+				errno = EACCES;
```

best not to fail loudly here.